CoreMapView/src/au/gov/asd/tac/constellation/views/mapview2/plugins/SelectOnGraphPlugin.java

```diff
@@ -0,0 +1,98 @@
+import au.gov.asd.tac.constellation.plugins.PluginInteraction;
+import au.gov.asd.tac.constellation.plugins.PluginType;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameters;
+import au.gov.asd.tac.constellation.plugins.templates.PluginTags;
+import au.gov.asd.tac.constellation.plugins.templates.SimpleEditPlugin;
+import java.util.ArrayList;
+import java.util.List;
+import org.openide.util.NbBundle;
+import org.openide.util.lookup.ServiceProvider;
+
+/**
+ * plugin to select nodes on the graph when there corresponding markers are
+ * selected in the map view
+ *
+ * @author altair1673
+ */
+@ServiceProvider(service = Plugin.class)
+@PluginInfo(pluginType = PluginType.SELECTION, tags = {PluginTags.SELECT})
+@NbBundle.Messages("SelectOnGraphPlugin=Highlights nodes on their corresponding marker gets selected on the map")
+public class SelectOnGraphPlugin extends SimpleEditPlugin {
+
+    // IDs of the selected nodes
+    private List<Integer> selectedNodeList = new ArrayList<>();
+    private boolean isSelectingVertex = true;
+
+    public SelectOnGraphPlugin(final List<Integer> selectedNodeList, final boolean isSelectingVertex) {
+        this.selectedNodeList = new ArrayList<>(selectedNodeList);
+        this.isSelectingVertex = isSelectingVertex;
+    }
+
+    public SelectOnGraphPlugin() {
+
+    }
+
+    /**
+     * Select vertices on graph which correspond to selected markers on the map
+     *
+     * @param graph
+     * @param interaction
+     * @param parameters
+     * @throws InterruptedException
+     * @throws PluginException
+     */
+    @Override
+    protected void edit(final GraphWriteMethods graph, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException, PluginException {
+
+        if (graph != null) {
+            // Select vertex
+            if (isSelectingVertex) {
+                final int vertexSelectedAttribute = VisualConcept.VertexAttribute.SELECTED.get(graph);
+                final int vertexCount = graph.getVertexCount();
+
+                for (int i = 0; i < vertexCount; ++i) {
+                    final int vertexID = graph.getVertex(i);
+                    graph.setBooleanValue(vertexSelectedAttribute, vertexID, selectedNodeList.contains(vertexID));
+                }
+
+                // Select transactions
+            } else {
+                final int transactionSelectedAttribute = VisualConcept.TransactionAttribute.SELECTED.get(graph);
+                final int transactionCount = graph.getTransactionCount();
+
+                for (int i = 0; i < transactionCount; ++i) {
+                    final int transactionID = graph.getTransaction(i);
+                    graph.setBooleanValue(transactionSelectedAttribute, transactionID, selectedNodeList.contains(transactionID));
+                }
+            }
+        }
+    }
+
+    @Override
+    public String getName() {
+        return "SelectOnGraphPlugin2";
```

Comment:
Maybe call this `Map View: Select on Graph` (that's not being used by the old map view)